src/chat-messages.js

```diff
@@ -0,0 +1,547 @@
+/**
+ * File: Chat Messages
+ *
+ * Maintainer:      - Sebastian Kippe <sebastian@kip.pe>
+ * Version:         - 0.5.0
+ *
+ * This module stores chat messages in daily archive files.
```

Comment:
Yes, it's supposed to cover all kinds of chat messages. Theoretically it already works and ircURI is optional and specific to IRC entries. You can give it any kind of `type` string when you instantiate the archive and it will use that in the archive URIs (e.g. `chat-messages/irc` could also be `chat-messages/slack` when you give it a type of "slack".

The ircURI is a bit clunky I think and should later be turned into something more generic like serverHost or something. Just couldn't think of something that sounds good enough yet and haven't had an opportunity to work with @elf-pavlik on it again, but it's fairly inconsequential to using it without setting hosts already.